File: sample-java/src/main/java/com/dylanc/tracker/sample/java/track/RecordThreadNode.java
package com.dylanc.tracker.sample.java.track;

import androidx.annotation.NonNull;

import com.dylanc.tracker.TrackNode;
import com.dylanc.tracker.TrackParams;

/**
 * @author Dylan Cai
 */
public class RecordThreadNode implements TrackNode {

  public boolean isRecord = false;

  @Override
  public void fillTackParams(@NonNull TrackParams params) {
    params.put("is_record", isRecord);
  }
}

File: sample-kotlin/src/main/java/com/dylanc/tracker/sample/track/RecordThreadNode.kt
package com.dylanc.tracker.sample.track

import com.dylanc.tracker.TrackNode
import com.dylanc.tracker.TrackParams
import java.io.Serializable

/**
 * @author Dylan Cai
 */
class RecordThreadNode : TrackNode, Serializable {
  var isRecord = false

  override fun fillTackParams(params: TrackParams) {
    params.put("is_record", isRecord)
  }
}